Sheffield | 25-SDC-Nov | Sheida Shabankari | Sprint 5 |Laptop allocation

sheida-shab opened the pull request:
- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

sheida-shab committed:
define people and laptops data & write basic  code to allocate laptops

sheida-shab committed:
improving codeand assign laptops to every person

sheida-shab committed:
refactoring code & fixing bugs

sheida-shab committed:
added some data

sheida-shab committed:
Added a few comments

sheida-shab committed:
Fixeing laptop os name in print

OracPrime commented on laptop_allocation.py:
```diff
@@ -0,0 +1,80 @@
+]
+
+# Global sadness counter
+sadness=0
```
Having a global variable is normally a bad idea.  This could just as easily be created inside the allocate_laptops function and returned as part of a multi-value return from the function.  As it is if you call the function twice the second time will include the sadness from the first time in its sadness score.

sheida-shab replied:
Thanks for your feedback.
I fixed it.

OracPrime commented on laptop_allocation.py:
```diff
@@ -0,0 +1,80 @@
+              if person.preferred_operating_system[i] == laptop.operating_system :
+                  allocated_history[person.name]=laptop # assign laptop
+                  sadness += i  # increment sadness by preference index
+                  laptops.remove(laptop)
```
The laptops argument is passed to allocate_laptops by reference.  So this remove function is modifying the original list of laptops.  Now this may be intentional, in which case it should be made obvious somewhere.  Alternatively make a copy of the list at the start of the function and modify that inside the function.  I can see arguments for both approaches being desirable, but it would be good to make it obvious.

OracPrime commented on laptop_allocation.py:
```diff
@@ -0,0 +1,80 @@
+            break     
+          
+     if not allocated_flag :   # assign any remaining laptop if preferred OS not found
+        allocated_history[person.name]=laptops[0]
```
Both here and above you're using person.name as they key in the dictionary.  On line 53 you declared the key as Person.  One or the other needs correcting.

OracPrime reviewed (commented):
Generally solid code, but a couple of bits need tidying up or clarifying.

sheida-shab committed:
Fixing the global variable issue

sheida-shab committed:
Adding comment about modifying laptop list

sheida-shab committed:
Fix dictionary key type to use person.name

sheida-shab committed:
Add a comment in .gitignore file

sheida-shab commented:
Hi David,

I just wanted to let you know that I’ve addressed all the points you mentioned in your previous comments. I noticed that you reviewed the PR again, but there weren’t any new comments added, so I’m not sure whether the PR is now considered complete or if there are still any issues to fix.

If possible, could you please take another look and let me know the current status?

Thank you very much for your time and support.

OracPrime commented:
Variable naming and readability are really important code quality issues, which is why I flagged it for one final fix

sheida-shab committed:
Rename variable sadness to total_sadness for more readability
